modules/pubsub/src/main/scala/zio/gcp/pubsub/subscriptionadmin/SubscriptionAdmin.scala

```diff
@@ -0,0 +1,199 @@
+package zio.gcp.pubsub.subscriptionadmin
+
+import zio._
+import java.util.concurrent.TimeUnit
+import com.google.cloud.pubsub.v1.SubscriptionAdminClient
+import com.google.cloud.pubsub.v1.stub.SubscriberStub
+import com.google.cloud.pubsub.v1.SubscriptionAdminSettings
+import com.google.pubsub.v1.CreateSnapshotRequest
+import com.google.pubsub.v1.Snapshot
+import com.google.pubsub.v1.ProjectSnapshotName
+import com.google.pubsub.v1.ProjectSubscriptionName
+import com.google.pubsub.v1.Subscription
+import com.google.pubsub.v1.DeleteSnapshotRequest
+import com.google.pubsub.v1.DeleteSubscriptionRequest
+import com.google.pubsub.v1.GetSubscriptionRequest
+import com.google.pubsub.v1.ListSnapshotsRequest
+import com.google.pubsub.v1.ProjectName
+import com.google.pubsub.v1.ListSubscriptionsRequest
+import com.google.pubsub.v1.ModifyPushConfigRequest
+import com.google.pubsub.v1.SeekResponse
+import com.google.pubsub.v1.SeekRequest
+import com.google.pubsub.v1.UpdateSnapshotRequest
+import com.google.pubsub.v1.UpdateSubscriptionRequest
+import com.google.iam.v1.{ Policy, SetIamPolicyRequest }
+import com.google.iam.v1.{ TestIamPermissionsRequest, TestIamPermissionsResponse }
+import com.google.iam.v1.{ GetIamPolicyRequest, Policy }
+import com.google.iam.v1.Policy
+import com.google.iam.v1.TestIamPermissionsResponse
+import com.google.pubsub.v1.TopicName
+import com.google.pubsub.v1.PushConfig
+
+object SubscriptionAdmin {
+
+  trait Service {
```

Comment:
I would expose the most common methods and provide a "hook" into the API, rather than writing many proxy methods.

Reply:
I'm not sure about this. It is yet to be determined what are the common methods.